Update workday docs - version 35 (#1064)
mulesoft/docs-connectors: workday/13.0/modules/ROOT/pages/workday-reference.adoc

= Workday Connector 13.0 Reference - Mule 4
:page-aliases: connectors::workday/workday-reference.adoc

Workday API version 35

Anypoint Connector for Workday (Workday Connector) provides access to standard Workday operations.

== Configurations
---
[[config]]
=== Configuration

==== Parameters
[%header%autowidth.spread]
|===
| Name | Type | Description | Default Value | Required
|Name | String | The name for this configuration. Connectors reference the configuration with this name. | | x

> | Connection a| * <<config_basic-authentication-api, Workday Basic Authentication Connection>> {nbsp}

Reviewer: ```suggestion
| Connection a| * <<config_basic-authentication-api, Workday Basic Authentication Connection>> 
```

Reviewer: Did you run the clean up script to remove these fields {nbsp}?  No worries now, for the next time.

|===

=== Connection Types

Connection authentication types to provide to this configuration.

[[config_basic-authentication-api]]
==== Workday Basic Authentication Connection

Uses your Workday username and password for authenticating the connection.

====== Parameters
[%header%autowidth.spread]
|======================
| Name | Type | Description | Default Value | Required
| Proxy Configuration a| One of:

* <<Default>>
* <<NTLM>> |  |  | {nbsp}
| Password a| String |  Password associated with the User Name |  | *x*{nbsp}
| Host name a| String |  Host name of your Workday instance |  `impl-cc.workday.com` | {nbsp}
| Tenant name a| String |  Workday tenant name or ID. Use expressions in this field to configure a connection for multiple tenants. |  | *x*{nbsp}
| User Name a| String |  User name |  | *x*{nbsp}
| Use Persistent Connections a| Boolean |  If false, each connection is closed after the first request is completed. |  true | {nbsp}

> | Max An expression that evaluates against the operation's output. The outcome of the expression is stored in the target variable. a| Number |  Maximum number of outbound connections to keep open concurrently. By default, the number of connections is unlimited. |  -1 | {nbsp}

Reviewer: ```suggestion
| Max Connections  a| Number |  Maximum number of outbound connections to keep open concurrently. By default, the number of connections is unlimited. |  -1 | {nbsp}
```

Reviewer: Found the same explanation down below for Max Connections

| User Name a| String |  The user name |  | *x*{nbsp}
| Use Persistent Connections a| Boolean |  If false, each connection is closed after the first request is completed. |  true | {nbsp}
| Max Connections a| Number |  Maximum number of outbound connections to keep open at the same time. By default, the number of connections is unlimited. |  -1 | {nbsp}
| Connection Idle Timeout a| Number |  Amount of time that a connection can remain idle before it is closed. The value of this attribute is used only when persistent connections are enabled. |  30 | {nbsp}
| Connection Idle Timeout Unit a| Enumeration, one of:

** NANOSECONDS
** MICROSECONDS
** MILLISECONDS
** SECONDS
** MINUTES
** HOURS
** DAYS |  Unit of time for Connection idle timeout |  SECONDS | {nbsp}
| Stream Response a| Boolean |  Whether or not to stream received responses, meaning processing continues as soon as all headers are parsed and the body is streamed as it arrives. When enabled, the response must eventually be read since, depending on the configured buffer size, it might not fit into memory and processing stops until space is available. |  false | {nbsp}

> | Response Buffer Size a| Number |  The space, in bytes, for the buffer where the HTTP response will be stored. |  -1 | {nbsp}

Reviewer: ```suggestion
| Response Buffer Size a| Number |  The space, in bytes, for the buffer where the HTTP response is stored. |  -1 | {nbsp}
```

* <<repeatable-file-store-stream>>
* non-repeatable-stream |  Configures how Mule processes streams. Repeatable streams are the default behavior. |  |
| Response Timeout a| Number |  How long to wait for a response, based on the unit in the Response Timeout field  |  |
| Response Timeout Unit a| Enumeration, one of:

** NANOSECONDS
** MICROSECONDS
** MILLISECONDS
** SECONDS
** MINUTES
** HOURS
** DAYS |  Time unit for the Response Timeout field |  |
| Target Variable a| String |  The name of a variable that stores the operation's output. |  |
| Target Value a| String |  An expression that evaluates against the operation's output. The outcome of the expression is stored in the target variable. |  `#[payload]` |
| Reconnection Strategy a| * <<reconnect>>
* <<reconnect-forever>> |  A retry strategy for connectivity errors. |  |
|===

==== Output
[%autowidth.spread]
|===
|Type |Binary
|Attributes Type | <<SoapAttributes>>
|===

=== For Configurations
* <<config>>

==== Throws
* WORKDAY:CONNECTIVITY
* WORKDAY:PROCESSING_ERROR
* WORKDAY:RETRY_EXHAUSTED
* WORKDAY:VALIDATION_ERROR

[[performanceManagement]]
=== Performance Management
`<workday:performance-management>`

> Invoke the Performance Management web service, which contains operations that expose Workday Employee Performance Management Business Services data. This web service can be used for integration with other employee performance management systems.

Reviewer: ```suggestion
Invoke the Performance Management web service, which contains operations that expose Workday Employee Performance Management Business Services data. Use this web service to integrate with other employee performance management systems.
```

> :page-aliases: connectors::workday/workday-reference.adoc

Reviewer 2: NO EDIT
